packages/vite/src/node/plugins/worker.ts

```diff
@@ -80,7 +95,9 @@ export function webWorkerPlugin(config: ResolvedConfig): Plugin {
             export default function WorkerWrapper() {
               const objURL = blob && (window.URL || window.webkitURL).createObjectURL(blob);
               try {
-                return objURL ? new Worker(objURL) : new Worker("data:application/javascript;base64," + encodedJs, {type: "module"});
+                return objURL ? new Worker(objURL) : new Worker("data:application/javascript;base64," + encodedJs, ${
+                  format === 'es' ? '{type: "module"}' : '{}'
+                });
```

Comment:
I think we should leave the previous fallback as is. That was already using `{type: "module"}` even when we were doing an `iife`. The logic should be in `new Worker(objURL)`, no?